README.md

```diff
@@ -1,133 +1,616 @@
+├── gradle.properties
+├── gradlew
+├── gradlew.bat
+├── models
+│   └── weather.smithy
+├── settings.gradle.kts
+└── smithy-build.json
+```
+
+To add a minimal `typescript-client-codegen` plugin, add the following to `smithy-build.json`: 
+
+```json
+// smithy-build.json
+{
+    "version": "1.0",
+    "sources": ["models"],
+    "plugins": {
+        // Add the Smithy TypeScript client plugin
+        "typescript-client-codegen": {
+            // Minimal configuration: add package name and version
+            "package": {
+                "package": "@smithy/typescript-example-client",
+                "packageVersion": "0.0.1"
+            }
         }
     }
+}
+```
 
-    plugins {
-        id("software.amazon.smithy").version("0.6.0")
-    }
+> Note: Maven dependencies cannot be configured in `smithy-build.json` for Gradle projects.
 
-    repositories {
-        mavenLocal()
-        mavenCentral()
-    }
+Then, add the `smithy-typescript-codegen` dependency in `build.gradle.kts`:
 
-    dependencies {
-        implementation("software.amazon.smithy:smithy-model:[1.33.0,1.34.0[")
-        implementation("software.amazon.smithy.typescript:smithy-typescript-codegen:0.18.0")
-    }
-   ```
-
-3. Create a `smithy-build.json` file with the following contents,
-   substituting "smithy.example#ExampleService" with the name of the service
-   to generate and "smithyexample" with the name of the TypeScript package to
-   create.:
-
-   ```json
-    {
-        "version": "1.0",
-        "plugins": {
-            "typescript-codegen": {
-                "service": "smithy.example#ExampleService",
-                "targetNamespace": "SmithyExample",
-                "package": "smithyexample",
-                "packageVersion": "0.0.1",
-                "packageJson": {
-                    "license": "Apache-2.0"
-                }
-            }
-        }
-    }
+```kotlin
+plugins {
+    id("java-library")
+    id("software.amazon.smithy.gradle.smithy-jar").version("0.8.0")
+}
 
-   ```
+repositories {
+    mavenLocal()
+    mavenCentral()
+}
 
-4. Create a directory named `model`. This is where all of your Smithy models
-   will go.
+dependencies {
+    val smithyVersion: String by project
 
-5. Create a file in the `model` directory named `main.smithy` with the following contents:
+    smithyCli("software.amazon.smithy:smithy-cli:$smithyVersion")
 
-    ```
-    $version: "2"
-   
-    namespace smithy.example
+    // Add the Smithy TypeScript code generator dependency
+    implementation("software.amazon.smithy.typescript:smithy-typescript-codegen:0.19.0")
 
-    service ExampleService {
-        version: "2022-01-01"
-        operations: [Echo]
-    }
+    // Uncomment below to add various smithy dependencies (see full list of smithy dependencies in https://github.com/awslabs/smithy)
+    // implementation("software.amazon.smithy:smithy-model:$smithyVersion")
+    // implementation("software.amazon.smithy:smithy-linters:$smithyVersion")
```

Comment:
We might want to have the linters added by default for a nice batteries-included experience.

Reply:
This would need to be an upstream change.